pallets/schemas/src/rpc/src/lib.rs

```diff
@@ -93,6 +94,20 @@ where
 		at: Option<<Block as BlockT>::Hash>,
 		schema: Vec<u8>,
 	) -> Result<FeeDetails<NumberOrHex>> {
+		// Validate serialized avro schema
+		let schema_string = String::from_utf8(schema.clone()).map_err(|_| RpcError {
+			code: ErrorCode::InvalidParams,
+			message: format!("invalid avro schema"),
+			data: Some(format!("{:?}", schema).into()),
+		})?;
+		let finger_print = avro::fingerprint_raw_schema(&schema_string);
+		if finger_print.is_err() {
```

Comment:
Run schema validator on incoming serialized schema before computing cost, this enable user to know if their schema is valid before registration and cost associated if valid